Add dpu flag so sonic-db-cli can make cli calls to smartswitch-only databases

croos12 opened the pull request:
sonic-db-cli does not currently support calling the dpu redis instances that exist on smartswitches. This changes adds a -d/--dpu flag to select a dpu redis instance for running commands so that dpu instances can be targeted by the sonic-db-cli from the host OS

croos12 committed:
Add dpu flag so sonic-db-cli can make cli calls to smartswitch-only databases

Signed-off-by: Connor Roos <croos@nvidia.com>

vivekrnv commented on common/dbconnector.cpp:
```diff
@@ -186,8 +186,9 @@ void SonicDBConfig::initialize(const string &file)
     if (m_init)
     {
-        SWSS_LOG_ERROR("SonicDBConfig already initialized");
-        throw runtime_error("SonicDBConfig already initialized");
```
Community will have a problem with this change. you can instead use reset()

Eg:
```
>>> SonicDBConfig::initializeGlobalConfig()
>>> SonicDBConfig.reset()
>>> SonicDBConfig.load_sonic_db_config('/var/run/redisdpu0/sonic-db/database_config.json')
>>> SonicDBConfig.getDbPort('CONFIG_DB')
6381
>>> SonicDBConfig.getDbHostname('CONFIG_DB')
'169.254.200.254'
```

vivekrnv commented:
Pleaase update description and mistakes in title

vivekrnv commented on sonic-db-cli/sonic-db-cli.cpp:
```diff
@@ -10,9 +10,27 @@
+
+void initializeConfig(const string& dpu_name = "")
+{
+    if(dpu_name.empty()){
```
Let's follow the formatting of current code

```
if(dpu_name.empty())
{

```

vivekrnv commented on sonic-db-cli/sonic-db-cli.cpp:
```diff
@@ -10,9 +10,27 @@
+    if(dpu_name.empty()){
+        SonicDBConfig::initialize(SonicDBConfig::DEFAULT_SONIC_DB_CONFIG_FILE);
+    }
+    else{
```
same as above

vivekrnv commented on sonic-db-cli/sonic-db-cli.cpp:
```diff
@@ -186,15 +206,16 @@ void parseCliArguments(
        {"help",        optional_argument, NULL,  'h' },
        {"unixsocket",  optional_argument, NULL,  's' },
        {"namespace",   optional_argument, NULL,  'n' },
+       {"dpu",    optional_argument, NULL,  'd' },
```
So, is unixsocket and dpu, since we are exclusively accessing DPU DB's over TCP

vivekrnv commented:
LGTM other than minor comments

ysmanman committed:
Add FIPS MACSec POST table in StateDB. (#1071)

saiarcot895 committed:
Skip kernel version check (#1078)

* Skip kernel version check

For CI pipeline purposes, loosen the requirement of needing an
exact/matching kernel source for the currently-running kernel version.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>

* Update wording

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>

---------

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>
Co-authored-by: Ze Gan <ganze718@gmail.com>

wen587 committed:
[strict yang] Add YANG based cfg_schema.h to schema.h (#1074)

Comparing to #1008, change to PYTHON3 for buster build failure.
Improve the dynamic debian_version as an argument instead of static bookworm.

Why I did it
To enable YANG based CFG table generation, the ConfigDB related macro will be generated by YANG model

How I did it
Add gen_cfg_schema.py script to generate cfg_schema.h which will be included in schema.h

Work item tracking
Microsoft ADO: 31676863
How to verify it
make deb build

croos12 committed:
Address comment and make flag work for any container

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Remove added imports

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Merge branch 'master' into users/croos/dpu-databases

croos12 committed:
Use getContainerFilePath helper method in container tests

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Merge remote-tracking branch 'refs/remotes/origin/users/croos/dpu-databases' into users/croos/dpu-databases

croos12 committed:
Use DPU_COUNTERS_DB for test

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Use output to test if container and path are correct.

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Add more print statements to better understand test failure

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Remove cli exception test to see if it is causing pipeline failure

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Try to catch test exception in pipeline

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Revert "Add more print statements to better understand test failure"

This reverts commit 4338bc460f556aeab6817b8b5a752cd45091e0f1.

croos12 committed:
Revert "Use output to test if container and path are correct."

This reverts commit 89a08b26316dad200fb306f7c9ba6c9962c5fde8.

croos12 committed:
Revert "Try to catch test exception in pipeline"

This reverts commit 34ad9bcc54e8ab471b9e3ef5a641748dfa02748a.

croos12 committed:
Revert "Remove cli exception test to see if it is causing pipeline failure"

This reverts commit 942cbb1064a591255a34fc346d3fe6fa145c3a15.

croos12 committed:
Revert "Use DPU_COUNTERS_DB for test"

This reverts commit 8b2451c6f572abffea105a887c322790c4c35d3d.

croos12 committed:
Change logic to add config_directory to any relative path, also add a test for coverage

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Use DPU_TEST_DB again and test argument exceptions in test_cli_run_dpu_cmd

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Test comparison with first line of help output for invalid arguments

Signed-off-by: Connor Roos <croos@nvidia.com>

croos12 committed:
Test for -1 exit codes for invalid input

Signed-off-by: Connor Roos <croos@nvidia.com>

saiarcot895 committed:
Fix trixie compilation and add Trixie to PR pipeline (#1069)

Fix a new compilation error on Trixie that popped up. Also, add a build for Trixie to the PR pipeline to prevent future errors.

qiluo-msft committed:
Add c-api/Rust wrappers for SonicV2Connector (#1080)

This PR adds c-api/Rust wrappers for SonicV2Connector and EventPublisher, so they will be available for Rust application development. Also add unit tests for c-api and Rust wrappers.

wen587 committed:
Prevent adding new CFG table since it is auto generated (#1081)

Add a checker to prevent adding new CFG_ macros when update common/schema.h.

Why I did it
CFG_XXX_TABLE_NAME macros should be auto-generated from YANG models via cfg_schema.h.

How I did it
Introduced .github/workflows/prevent-cfg.yml which runs on PRs that modify common/schema.h.
It extracts all CFG_ macros defined in the file and compares them against an allowed legacy list.

Work item tracking
Microsoft ADO: 35075048

croos12 committed:
Merge branch 'master' into users/croos/dpu-databases
